src/lib/EngineOptionsFactory.ts

```diff
@@ -28,7 +28,9 @@ abstract class CommonEngineOptionsFactory implements EngineOptionsFactory {
 		const options: Map<string,string> = new Map();
 		if (this.shouldSfgeRun(inputs)) {
 			const sfgeConfig: SfgeConfig = {
-				projectDirs: this.inputProcessor.resolveProjectDirPaths(inputs)
+				projectDirs: this.inputProcessor.resolveProjectDirPaths(inputs),
+				cachepath: inputs.cachepath,
+				enablecaching: inputs.enablecaching
```

Comment:
The non-DFA command doesn't use these options, right? So they probably shouldn't be declared here in the Common factory, and should probably just be added in the DFA factory the way properties like `pathexplimit` are.